rotkehlchen/api/v1/schemas.py

```diff
@@ -2026,3 +2023,95 @@ class ReverseEnsSchema(AsyncIgnoreCacheQueryArgumentSchema):
     location_data_snapshot_file = FileField(allowed_extensions=['.csv'], required=True)
+
+
+class SnapshotQuerySchema(SnapshotTimestampQuerySchema):
+    action = fields.String(
+        validate=webargs.validate.OneOf(choices=('export', 'download')),
+        load_default=None,
+    )
+    path = DirectoryField(load_default=None)
+
+    @validates_schema
+    def validate_schema(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        path = data.get('path')
+        action = data.get('action')
+        if action == 'export' and path is None:
+            raise ValidationError(
+                message='A path has to be provided when action is export',
+                field_name='action',
+            )
+
+
+class BalanceSnapshotSchema(Schema):
+    timestamp = TimestampField(required=True)
+    category = SerializableEnumField(enum_class=BalanceType, required=True)
+    asset_identifier = AssetField(required=True, form_with_incomplete_data=True)
+    amount = AmountField(required=True)
+    usd_value = AmountField(required=True)
+
+    @post_load
+    def make_balance(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> Dict[str, Any]:
+        return {
+            'timestamp': int(data['timestamp']),
+            'category': data['category'].serialize(),
+            'asset_identifier': str(data['asset_identifier'].identifier),
+            'amount': str(data['amount']),
+            'usd_value': str(data['usd_value']),
+        }
+
+
+class LocationDataSnapshotSchema(Schema):
+    timestamp = TimestampField(required=True)
+    location = LocationField(required=True)
+    usd_value = AmountField(required=True)
+
+    @post_load
+    def make_location_data(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> Dict[str, Any]:
+        return {
+            'timestamp': data['timestamp'],
+            'location': data['location'].serialize(),
+            'usd_value': str(data['usd_value']),
+        }
+
+
+class SnapshotEditingSchema(Schema):
+    timestamp = TimestampField(required=True)
+    balances_snapshot = fields.List(
+        fields.Nested(BalanceSnapshotSchema),
+        required=True,
+        validate=webargs.validate.Length(min=1),
+    )
+    location_data_snapshot = fields.List(
+        fields.Nested(LocationDataSnapshotSchema),
+        required=True,
+        validate=webargs.validate.Length(min=1),
+    )
+
+    @validates_schema
+    def validate_schema(  # pylint: disable=no-self-use
+        self,
+        data: Dict[str, Any],
+        **_kwargs: Any,
+    ) -> None:
+        if not data['timestamp'] == data['balances_snapshot'][0]['timestamp'] == data['location_data_snapshot'][0]['timestamp']:  # noqa: 501
+            raise ValidationError("timestamp does not match")
```

Comment:
Question. Here should the timestamp match for all the entries in the list?

Reply:
Yes